### ISSUE TITLE
Connection refused

### ISSUE THREAD

**Issue by KreAch3R** [KrystianD/bm2-battery-monitor#16]
Hello, and thanks for this awesome repo!

I tried your code at my device and I get `[Errno 111] Connection refused.` errors. I assume this is because the `KEY` changed? Can you help?

> [2023-02-28 01:26:47] A message handler raised an exception: [Errno 111] Connection refused.
Traceback (most recent call last):
  File "src/dbus_fast/message_bus.py", line 811, in dbus_fast.message_bus.BaseMessageBus._process_message
  File "/home/kreach3r/.local/lib/python3.9/site-packages/bleak/backends/bluezdbus/manager.py", line 860, in _parse_msg
    on_value_changed(message.path, self_interface["Value"])
  File "/home/kreach3r/.local/lib/python3.9/site-packages/bleak/backends/bluezdbus/client.py", line 152, in on_value_changed
    callback(bytearray(value))
  File "/home/kreach3r/bm2-battery-monitor/bm2_mqtt/bm2_mqtt.py", line 36, in notification_handler
    single(payload=payload, topic=mqtt_topic, port=mqtt_port, hostname=mqtt_hostname)
  File "/home/kreach3r/.local/lib/python3.9/site-packages/paho/mqtt/publish.py", line 240, in single
    multiple([msg], hostname, port, client_id, keepalive, will, auth, tls,
  File "/home/kreach3r/.local/lib/python3.9/site-packages/paho/mqtt/publish.py", line 176, in multiple
    client.connect(hostname, port, keepalive)
  File "/home/kreach3r/.local/lib/python3.9/site-packages/paho/mqtt/client.py", line 914, in connect
    return self.reconnect()
  File "/home/kreach3r/.local/lib/python3.9/site-packages/paho/mqtt/client.py", line 1044, in reconnect
    sock = self._create_socket_connection()
  File "/home/kreach3r/.local/lib/python3.9/site-packages/paho/mqtt/client.py", line 3685, in _create_socket_connection
    return socket.create_connection(addr, timeout=self._connect_timeout, source_address=source)
  File "/usr/lib/python3.9/socket.py", line 843, in create_connection
    raise err
  File "/usr/lib/python3.9/socket.py", line 831, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 111] Connection refused
^CTraceback (most recent call last):
  File "/home/kreach3r/bm2-battery-monitor/bm2_mqtt/bm2_mqtt.py", line 83, in <module>
    loop.run_until_complete(main())
  File "/usr/lib/python3.9/asyncio/base_events.py", line 629, in run_until_complete
    self.run_forever()
  File "/usr/lib/python3.9/asyncio/base_events.py", line 596, in run_forever
    self._run_once()
  File "/usr/lib/python3.9/asyncio/base_events.py", line 1854, in _run_once
    event_list = self._selector.select(timeout)
  File "/usr/lib/python3.9/selectors.py", line 469, in select
    fd_event_list = self._selector.poll(timeout, max_ev)
KeyboardInterrupt

**Comment by KrystianD** [comment]
Hi @KreAch3R, thanks!

AES key doesn't play any role in connection establishing process, so this is something else, unfortunately, due to amount of different devices, I am unable to help.

**Comment by KreAch3R** [comment]
Thanks for the super fast reply!

Are you talking about the battery monitor device? I have the BM2 app and Battery Monitor, as close to official as it can get with chinese products. So it works in your case with the latest app?

**Comment by KreAch3R** [comment]
Ok, I think I figured it out. I needed to install a `local MQTT broker` as per [here](https://stackoverflow.com/a/60014986/4008886). After I installed mosquitto as per [here](https://www.howtoforge.com/how-to-install-mosquitto-mqtt-message-broker-on-debian-11/), and run the script using `python3`, I get:
`[2023-02-28 03:15:15] Connected`

But nothing else prints. I assume it's waiting on the device to send something? I have it currently connected on a test power supply and it's only giving out 11,6V, is that the reason it doesn't output anything?
